File: web/config/brand.js
'use strict';

/**
 * web/config/brand.js
 *
 * SINGLE SOURCE OF TRUTH for the product brand (name + logo) across the WHOLE web
 * UI. Change it HERE and it updates everywhere: page <title>s, the sidebar badge,
 * the login / forgot screens, invoice print footer, the PWA manifest, and JS
 * toasts (via window.BRAND, injected in _layout.ejs).
 *
 * To rebrand:
 *   • name / shortName / tagline / color  → edit the values below.
 *   • the logo image                       → replace web/public/img/logo.svg
 *                                            (keep the same path).
 *   • the brand chip icon                  → change iconClass (any Font Awesome).
 *
 * This is ONE of FOUR brand files — one per runtime, because they can't share
 * code: this one, api/config/brand.js, agent/brand.py, and app/lib/core/brand.dart.
 * They must all agree. api/tests/brandConsistency.test.js pins name/tagline
 * across all four and fails loudly if one is changed without the others.
 */

module.exports = {
    // Full product name — shown in the UI and in the "<page> · <name>" titles.
    name:      'Teloora',
    // Compact form for tight spaces / the PWA short_name.
    shortName: 'Teloora',
    // One-line description (meta description + PWA description).
    tagline:   'Connected Accounting',
    // The square logo mark (favicon + sidebar chip). Replace the file to swap.
    logo:      '/img/logo.svg',
    // Same mark as a raster, for places that can't render SVG.
    logoMark:  '/img/logo-mark.png',
    // The full lock-up (wordmark + tagline) — login / forgot / PDF letterhead,
    // i.e. anywhere it renders big enough for the tagline to be readable.
    logoFull:  '/img/logo-full.png',
    // Wordmark only, no tagline. For tight chrome like the sidebar header,
    // where dropping the strapline lets the name itself run much larger.
    logoWordmark: '/img/logo-wordmark.png',
    // Fallback Font Awesome icon, used only if the logo image fails to load.
    iconClass: 'fa-solid fa-diagram-project',
    // Primary accent (chip gradient / theme). One knob for the brand colour.
    color:     '#1560E0',
    // Full brand palette — the logo's blue→green gradient and its deep navy
    // wordmark colour. Used by theme.css, the PDF/letterhead footer, and
    // anywhere else that needs more than a single accent.
    colors: {
        navy:     '#17265E',
        blue:     '#1560E0',
        green:    '#45B649',
        gradient: 'linear-gradient(135deg, #1560E0, #45B649)',
    },
    // UI theme knobs derived from the logo. These are emitted as CSS custom
    // properties by web/views/partials/brand-vars.ejs (included on every page,
    // after theme.css) so the whole UI follows the logo from this one place.
    theme: {
        primary:     '#1560E0',  // logo blue  → buttons, links, active states
        primaryDark: '#0F4CB8',  // hover/pressed shade of the blue
        secondary:   '#45B649',  // logo green → gradient end, accents
        // NOTE: the sidebar's own colours are NOT here. They are app chrome,
        // not brand identity, and live in theme.css (--sidebar-bg and friends).
    },
};

File: app/lib/core/brand.dart
/// core/brand.dart
///
/// SINGLE SOURCE OF TRUTH for the app's product brand (name + logo). Change it
/// HERE and it updates across the whole app: the MaterialApp title, the splash /
/// login / forgot screens, the update dialog, the sync-agent hint, and the
/// Settings default app name.
///
/// To rebrand:
///   • name / shortName / tagline  → edit the constants below.
///   • the logo image              → replace assets/images/logo.svg (same path).
///   • the launcher icon           → replace assets/icon/app_icon.png.
///
/// This is ONE of FOUR brand files — one per runtime, because they can't
/// share code: this one, web/config/brand.js, api/config/brand.js, and
/// agent/brand.py. They must all agree. api/tests/brandConsistency.test.js
/// pins name/tagline across all four and fails loudly if one is changed
/// without the others.
class Brand {
  Brand._();

  /// Full product name shown in the UI.
  static const String name = 'Teloora';

  /// Compact form for tight spaces.
  static const String shortName = 'Teloora';

  /// One-line description.
  static const String tagline = 'Connected Accounting';

  /// The single logo asset (splash / login). Replace the file to swap the logo.
  static const String logoAsset = 'assets/images/logo.svg';

  /// The full lock-up (mark + wordmark + tagline) — the same PNG the web
  /// sign-in / forgot pages show on their mobile breakpoint
  /// (web/public/img/logo-full.png). Keep the two files in step.
  static const String logoFullAsset = 'assets/images/logo-full.png';

  /// The launcher / app icon asset.
  static const String iconAsset = 'assets/icon/app_icon.png';

  /// Primary accent (matches web/config/brand.js color). One knob for the brand colour.
  static const String colorHex = '#1560E0';

  /// Full brand palette — the logo's blue→green gradient and its deep navy
  /// wordmark colour.
  static const String navyHex = '#17265E';
  static const String blueHex = '#1560E0';
  static const String greenHex = '#45B649';
}
